op-node/rollup/derive/test/random.go

```diff
@@ -0,0 +1,23 @@
+package test
+
+import (
+	"math/rand"
+
+	"github.com/ethereum-optimism/optimism/op-node/eth"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/ethereum-optimism/optimism/op-node/testutils"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/trie"
+)
+
+// RandomL2Block returns a random block whose first transaction is a random
+// Deposit transaction.
+func RandomL2Block(rng *rand.Rand, txCount int) (*types.Block, []*types.Receipt) {
+	l1Block := types.NewBlock(testutils.RandomHeader(rng),
+		nil, nil, nil, trie.NewStackTrie(nil))
+	l1InfoTx, err := derive.L1InfoDeposit(0, l1Block, eth.SystemConfig{}, false)
```

Comment:
I think this works. It's probably a little tricky to move the `L1InfoDeposit` to a base package

Reply:
Yes there's a 🐰 🕳️ of moving some types & functions into different packages as to not have to import `rollup/derive` everywhere. But that seemed too much to handle in this PR 😅